morituri/common/program.py

```diff
@@ -465,7 +456,9 @@ def getTagList(self, number):
             if self.metadata.release is not None:
                 tags['DATE'] = self.metadata.release
 
-            if number > 0:
+            has_mbinfo = mbidTrack and mbidTrackArtist and mbidAlbum \
+                    and mbidTrackAlbum and mbDiscId
+            if number > 0 and has_mbinfo:
```

Comment:
Would it be worth assigning tags['...TRACKID'] et al. within individual if branches instead of all at once? I'm not sure if it's useful or not to have, say, ARTISTID and not ALBUMID, or if that would ever happen, but it might be worth it? *shrugs*

Reply:
Nope, this `if` is protecting the insert of MusicBrainZ tags, and you absolutely don't want them when using `--allow-freedb`. This code basically checks for every field that is supposed to be present with MusicBrainZ; if any is missing, that's because we're hacking FreeDB stuff in it and so you don't want to end up with broken MusicBrainZ references obtained from dubious sources.